toolchain/sem_ir/stringify.cpp

```diff
@@ -420,10 +420,14 @@ class Stringifier {
   auto StringifyInst(InstId /*inst_id*/, ImplWitnessAccess inst) -> void {
     auto witness_inst_id =
         sem_ir_->constant_values().GetConstantInstId(inst.witness_id);
-    if (auto specific_interface =
-            TryGetSpecificInterfaceForImplWitness(witness_inst_id)) {
-      const auto& interface =
-          sem_ir_->interfaces().Get(specific_interface->interface_id);
+    auto specific_interface =
+        TryGetSpecificInterfaceForImplWitness(witness_inst_id);
```

Comment:
Cool, then we can get rid of the whole optional:

```suggestion
    auto lookup = sem_ir_->insts().GetAs<LookupImplWitness>(witness_inst_id);
    auto specific_interface = sem_ir_->specific_interfaces().Get(
          lookup.query_specific_interface_id);
```

And `TryGetSpecificInterfaceForImplWitness` can be removed.